### ISSUE TITLE
Occasional Incorrect Start Position

### ISSUE THREAD

**Issue by Jazcash** [Jazcash/sdfz-demo-parser#20]
Randomly decided to investigate what's causing this bug more closely, which leads to incorrect start positions showing on the website:

![image](https://github.com/user-attachments/assets/497d7cd1-eca7-44fd-af99-ba7ea2731337)

I believe the culprit stems from this section of engine code: https://github.com/beyond-all-reason/spring/blob/2ef0794f99d9215f44c07ec7aae57fe8c4600caa/rts/Net/NetCommands.cpp#L517-L528

The ingame UI doesn't allow players to place outside of their start box, but clients still send STARTPOS commands anyway, and the engine clamps/ignores them here.

Because the demo parser doesn't know the exact dimensions of the map, along with start box sizes always being on a scale of 0-200, but with STARTPOS position data being absolute map locations - I don't think this problem is solvable without a game or engine change.

Invalid start pos placement could be caught client side before they're sent over the network, but there could still be oddities with the final placement, e.g. if the player doesn't pick a spot at all and the game decides for them. So unfortunately I think the best solution to this problem is writing a gadget to `SendLuaRulesMsg` the one, true spawn position of each player's commander once the game starts, so it can be parsed from the replay.

**Comment by fcaps** [comment]
Another Update: 
it looks like that we cannot trust the STARTPOS at all, even the final one from the server before game_start.
For making this reliable we have to apply all the validations that the client does.

- its the players pick (drafmode)
- not in 350elmo of another player
- in the allyTeam Startbox
- on traversable ground
- player is not locked
- maybe even more

since this is a "state" and not a command anymore, i would opt to write a LUAMGS as OP suggested with the startposition,
or making Server and Client behave the same by adding the validation:

Client:
https://github.com/beyond-all-reason/RecoilEngine/blob/8ceb1c0e852a0490ad05724b567231127e3f4e02/rts/Net/NetCommands.cpp#L524

Server:
https://github.com/beyond-all-reason/RecoilEngine/blob/8ceb1c0e852a0490ad05724b567231127e3f4e02/rts/Net/GameServer.cpp#L1196

Not into the game and what that would mean, but surely this can help somebody else who is into all of this.

**Comment by p2004a** [comment]
Makes sense, you need you ask in `#dev-main` on Discord, there are literally tens of core game devs that should be able to help directly or point you at better direction (I don't contribute to core game). If nobody replies, ping me and I will make noise.

**Comment by fcaps** [comment]
i tried it 2-3 times to get some help, but this is not sexy enough. 
Sadly i have to implement a fallback anyways and focus my time there to correct the startpos with a "best guess". 

Here is a draft-pr that works when i am adding it to my local game and start a game with bots (cant really test it with real players, since not sure about on how to spin up multiple game instances and play a game online).
https://github.com/beyond-all-reason/Beyond-All-Reason/pull/6940
